help/en/html/apps/DecoderPro/TransformDecoder.shtml

```diff
@@ -0,0 +1,402 @@
+          all generated parts.
+      </p>
+      <p>
+          To give some example of simplification possible - let's take the decoder file
+          <code>Public_Domain_dccdoma_ARD_SCOM_MX.xm</code>. It configures a decoder, capable
+          of displaying signal aspects on several signal masts. The configuration contains
+          over 500 of CVs - yet the basic idea behind the configuration is dead simple:
+     </p>
+          <ul>
+            <li>a default aspect for each signal mast</li>
+            <li>for each signal mast AND for each one of 32 possible aspects, the number of signal to be displayed, 
+                interpreted by the decoder itself</li>
+          </ul>
+     <p>
+          A few statistics:
+     </p>
+          <ul>
+              <li>original decoder's definition: <b>870 kByte</b> 20608 lines.</li>
+              <li>stylesheet file: <b>12 kByte</b>, 257 lines.</li>
+              <li>decoder file template: <b>18 kByte</b>, 390 lines.</li>
+          </ul>
+     <p>
+          For JMRI itself or the speed of DecoderPro operation, these two approaches are the same: the file template
+          is internally transformed (expanded) to the decode definition XML and processed as if it was written
+          entirely by hand. For <b>maintenance</b>, it is a way easier to maintain ~600 lines of XML than 20600.
+     </p>
+     <p>
+          JMRI provides an option to apply a <b>XSLT stylesheet</b> to a decoder file,
+          <b>before</b> the file is loaded into DecoderPro and before it is interpreted
+          as CV variables and panels. This allows to hand-write unique CV definitions
+          and their panes, and <b>add generated</b> content where appropriate. 
+      </p>
+      <h2>
+          Example files
+      </h2>
+      <p>
+          To illustrate the techniques described here, a few example files are provided; all the files are licensed under
+          GNU GPL.
+      </p>
+      <ul>
+          <li><b><a href="resources/decoder-template.xml">decoder-template.xml</a></b> - the decoder definition <b>template</b> 
+          <li><b><a href="resources/scom.xsl">scom.xsl</a></b> - the stylesheet</li>
+      </ul>
+      <p>
+          The decoder template should be placed into the <b>xml/decoders</b> folder of the JMRI installation. It is <b>based on Petr Sidlo's
+          dccdoma.cz - ARD-SCOM-MX decoder</b> - generates the same decoder panels as the original one (as of 12/2019). The stylesheet
+          (<b>scom.xsl</b>) should be placed also into <b><code>xml/decoder</code></b> folder of the JMRI installation.
+      </p>
+      <p>
+          The template can be processed from the commandline to generate the decoder XML, so you can inspect effects of changing the
+          stylesheet and/or data embedded in the decoder template. The commandline for Linux:
+      </p>
+      <pre>
+xsltproc scom.xsl decoder-template.xml > decoder-gen.xml
```

Comment:
Should this mention you have to be in the xml/decoders directory for that line to work?  Or it could be `xsltproc xml/decoders/scom.xsl xml/decoders/decoder-template.xml > decoder-gen.xml

Reply:
Oops, just saw the paragraph later. Never mind.